utils/iterators.py

```diff
@@ -2,19 +2,10 @@ class RandomKeyIterator:
 
     def __init__(self, keys: list):
         self.keys = list(set(keys))
-        self.visited = list()
```

Comment:
I'm not sure what your plans are for self.visited. If it's only purpose is a placeholder to reset the self.keys array, then this commit condenses the class down quite a bit.